docs/site/submitting_a_pr.md

```diff
@@ -0,0 +1,629 @@
+
+Before squashing commits, ensure that you have performed the `rebase` steps described in the previous section.
+
+Ensure that you currently have your feature branch checked out.
+
+Type :
+
+```
+git status
+```
+
+![submit_pr_squash_commits_2.png](./imgs/submit_pr_squash_commits_2.png)
+
+I do have my feature branch checked out.
+
+Now let's rebase the feature branch off of the master branch in an interactive mode that allows us to `squash` the commits.
+```
+git rebase -i master
+```
+
+![submit_pr_squash_commits_3.png](./imgs/submit_pr_squash_commits_3.png)
+
+The `vim` editor opens and lists my 3 commits, and some commands that are available.
+
+Type `i` to place the editor into `INSERT` mode.
+
+![submit_pr_squash_commits_4.png](./imgs/submit_pr_squash_commits_4.png)
+
+I want to keep the commit message from the first commit, so I will leave the word `pick` in front of the first commit, and change the words from `pick` to `squash` for the second and third commits. (Use the `arrow` keys to move around, and the `delete` or `backspace` key for deleting characters)
+
+![submit_pr_squash_commits_5.png](./imgs/submit_pr_squash_commits_5.png)
+
+To save these changes, press the `escape` key, and type:
+
+```
+:wq
+```
+![submit_pr_squash_commits_6.png](./imgs/submit_pr_squash_commits_6.png)
+
+The `vim` editor comes up again to give me a chance to change my commit message.
+You'll notice that because I previously chose `squash` for the second and third commits, it appended the commit messages of the second and third commits onto the commit message of the first.
+
+![submit_pr_squash_commits_7.png](./imgs/submit_pr_squash_commits_7.png)
+
+This can give a user a chance to view all messages before forumating one final message.
+
+If I had a large number of commits and didn't want all commit messages to be appended together, I would have specified: `pick, fixup, fixup` instead of `pick, squash, squash` . The command `fixup` is like `squash`, except that it discards a commit's log message.
+
+In my case, I only like the message of the first commit, so I will delete the others.
+
+Press `i` to place the editor into `INSERT` mode, and prepare the single commit message.
+
+![submit_pr_squash_commits_8.png](./imgs/submit_pr_squash_commits_8.png)
+
+To save these changes, press the `escape` key, and type:
+
+```
+:wq
+```
+
+The interactive `rebase` command finally completes.
+
+![submit_pr_squash_commits_9.png](./imgs/submit_pr_squash_commits_9.png)
+
+
+Now it is time to push the changes from the local feature branch to the remote feature branch.
+
+Type :
+
+```
+git push -f
+```
+
+You can confirm that there is only 1 commit by typing:
+
+```
+git log
+```
+
+![submit_pr_squash_commits_11.png](./imgs/submit_pr_squash_commits_11.png)
+
+or via your browser in your remote feature branch.
+
+![submit_pr_squash_commits_10.png](./imgs/submit_pr_squash_commits_10.png)
+
+
+### 17. Creating the pull request (PR)
```

Comment:
PR can be created before squashing commits, which is usually the final step before the PR is ready to merge.